src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,127 @@
 import React from 'react';
 import './TodoList.scss';
+import { getTodos } from '../../api/api';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  selectedUserId: number,
+  onUserChange: (userId: number) => void,
+};
+type State = {
+  todos: Todo[],
+  filterByStatus: string,
+  titleFilter: string,
+  prevStatus: string,
+  prevTitle: string,
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state: State = {
+    todos: [{
+      id: 0,
+      createdAt: '',
+      updatedAt: '',
+      userId: 0,
+      title: '',
+      completed: false,
+    }],
+    filterByStatus: 'all',
+    titleFilter: '',
+    prevStatus: '',
+    prevTitle: '',
+  };
+
+  async componentDidMount() {
+    this.setState({ todos: await getTodos() });
+  }
+
+  componentDidUpdate() {
+    if (this.state.filterByStatus !== this.state.prevStatus
+      || this.state.prevTitle !== this.state.titleFilter) {
+      this.handleStatusFilter();
+    }
+  }
```

Comment:
`componentDidUpdate()` takes as its first two arguments the previous props and the previous state. Read more https://dev.to/cesareferrari/how-to-use-componentdidupdate-in-react-30en